### PR TITLE
Fix: additional model tags error and attribution error on bento._info

### PR DESCRIPTION
**Pull request by parano**
## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/types.py [hunk 1]
```diff
@@ -89,8 +89,6 @@ def from_taglike(cls, taglike: t.Union["Tag", str]) -> "Tag":
             return taglike
         return cls.from_str(taglike)
 
-        raise TypeError(f"can't make a Tag from {type(taglike)}")
```
Comment (parano) [inline bentoml/_internal/types.py:92]: @sauyon what was the intention of this line? ok to remove?

File: bentoml/_internal/bento/bento.py [hunk 2]
```diff
@@ -79,12 +79,12 @@ def _Bento_set_fs(bento: "Bento", _: t.Any, new_fs: "FS") -> "FS":
     return new_fs
 
 
-@attr.define(repr=False)
+@attr.define(repr=False, auto_attribs=False)
```
Comment (parano) [inline bentoml/_internal/bento/bento.py:82]: @sauyon `bentoml.get` currently raise an AttributionError and failed to access `_info`, the same happens to `_doc`. setting `auto_attribs=False` solved the issue but I'm not entirely sure why that was happening. seem like a bug in attrs to me.

File: bentoml/sklearn.py [hunk 3]
```diff
@@ -36,14 +36,10 @@ def _get_model_info(
...
     if model.info.module != __name__:
-        if model.info.module == "bentoml.mlflow":
-            pass
-        else:
-            raise BentoMLException(  # pragma: no cover
-                f"Model {tag} was saved with module"
-                f" {model.info.module}, failed loading"
-                f" with {__name__}."
-            )
+        raise BentoMLException(  # pragma: no cover
```
Comment (parano) [inline bentoml/sklearn.py:39]: cc @aarnphm - this check should be removed right?

Comment (aarnphm) [reply bentoml/sklearn.py:39]: we can just have it so that users can't load untrusted pickle file, since for sklearn we don't even use sklearn at all

Comment (parano) [reply bentoml/sklearn.py:39]: We still have the module check, I mean the removed part that allows loading bentoml.mlflow saved model with bentoml.sklearn module

Comment (aarnphm) [reply bentoml/sklearn.py:39]: oh it should be removed in some of the other PR i had

**Pull request merged**